### ISSUE TITLE
Improve SOLR testing, test against SOLR instead of mocking.

### ISSUE THREAD

**Issue by Rotonen** [4teamwork/opengever.core#4844]
_Harvest: `OneGov GEVER` - `Softwarepflege (Refactoring/Bug)`_

_meta: invest the assigned days_

We already have the capability of testing agains a mocked SOLR.

See 
https://github.com/4teamwork/opengever.core/blob/1cff81b94320be2512a27841f1ea4c7d0f50f1c3/opengever/base/tests/test_search.py#L185
for details.

Candidates for improved testing:
1) Livesearch
2) Tabbedview
3) Any listings
4) live search endpoint

**Comment by Rotonen** [comment]
Wishlist feature: have the integration testing fixtures automatically generate mock SOLR responses.

**Comment by Rotonen** [comment]
And we should probably also end up in a state where https://github.com/4teamwork/opengever.core/pull/5163 does not happen again.

**Comment by njohner** [comment]
Note that I implemented some `Livesearch` tests while working on #5018 (PR incoming)

**Comment by deiferni** [comment]
Up for discussion: also testing against a real Solr may be a good idea, already done in https://github.com/4teamwork/ivbe.intranet/blob/master/ivbe/intranet/testing.py#L61. Thanks @elioschmutz for the hint/example.

**Comment by Rotonen** [comment]
@deiferni @elioschmutz does that solr setup/teardown approach mean we cannot split that layer to run in parallel of itself as it'll conflict on both the port allocation and the filesystem resources it uses?

**Comment by elioschmutz** [comment]
Parallel tests with this approach will be tricky.

I think we need to share the solr-instance but using one core for each testing-process. But this is going beyond my solr-knowledge.

The easier way would be to put all solr-layer tests within one proccess...

**Comment by Rotonen** [comment]
@elioschmutz that'll still break with a static port allocation used on the layer, if multiples of the same test happen to run on the same server at the same time. With `opengever.core` this is fairly likely.

**Comment by Rotonen** [comment]
I had a brief chat with @buchi and the way to go here is to have a shared Solr installation (per version of Solr we use) deployed to the CI infrastructure and have the test layers just create new cores with unique names over the REST API of Solr.

This is thus currently pending work by operations on the CI infrastructure.

**Comment by Rotonen** [comment]
The shuffle nightlies have brought to attention a further caveat of the current status - it is currently very easy to do test class level leaks across tests as the tests manipulate `self.connection`.

https://github.com/4teamwork/opengever.core/blob/510047189f0437bb7f6e442b3309ec8e1d2611ed/opengever/api/tests/test_listing.py#L286-L300

Output of the failure in question:

```
Error in test test_filter_by_start_date (opengever.api.tests.test_listing.TestListingEndpointWithSolr)
Traceback (most recent call last):
  File "/Users/joni/.homebrew/var/pyenv/versions/2.7.15/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/Users/joni/.buildout/eggs/cp27m/ftw.testbrowser-1.30.1-py2.7.egg/ftw/testbrowser/__init__.py", line 42, in test_function
    return func(self, *args, **kwargs)
  File "/Users/joni/koodaus/git/opengever.core/opengever/api/tests/test_listing.py", line 296, in test_filter_by_start_date
    filters = self.conn.search.call_args[0][0]['filter']
TypeError: 'NoneType' object has no attribute '__getitem__'
```

So whatever the other tests in that class (or in Solr mocking in general) do, either has to have happened before running that test, or even worse, the current status relies on the run order by relying on leaks having happened.

For local reproducibility:
`Tests were shuffled using seed number 397596307423.`
`bin/test --shuffle --shuffle-seed=397596307423 -x`

**Comment by phgross** [comment]
Duplicate of #5797
